### ISSUE TITLE
Feature: options to ignore try,catch for rule global-require

### ISSUE THREAD

**Issue by evilebottnawi** [eslint/eslint#3834]
Example:
```javascript
try {
    posix = require('posix');
} catch (ex) {
    // Nothing
}
```
Now i get error global-require

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by btmills** [comment]
This pattern is explicitly disallowed in the docs, but I've seen it before when trying to import an optional dependency. @evilebottnawi what would such an option look like?

**Comment by evilebottnawi** [comment]
@btmills i see two ways import optional dependency.
First: 
```javascript
try {
    posix = require('posix');
} catch (ex) {
    // Nothing
}
```
Second: 
```javascript
try {
    var posixPath = require.resolve('posix');

    if (posixPath) {
        posix = require('posix');
    }
} catch (exception) {
    // Nothing
}
```
I think first way is better, less code and faster.

**Comment by btmills** [comment]
```js
var moduleName = 'preferred';
try {
    require.resolve(moduleName);
} catch (err) {
    moduleName = 'fallback';
}
var optionalModule = require(moduleName);
```

That works today, but it's ugly.

**Comment by evilebottnawi** [comment]
@btmills What is required is not always the fallback

**Comment by nzakas** [comment]
I'm not sure an exception makes sense here. It seems like a very special case that might best be handled via configuration comments.

**Comment by evilebottnawi** [comment]
@nzakas maybe then it can be put in the documentation

**Comment by nzakas** [comment]
What would you suggest be put in the documentation?

**Comment by jamestalmage** [comment]
If nothing else the documentation should use a different word than "scope"

> Disallow require() outside of the top-level module **scope**

To me that infers variable scope, but most of the `warn` examples have nothing to do with variable scope (`try` blocks included).

The argument regarding the performance implications of require only apply if it is **actually** in a nested variable scope (a function which can be deferred).

I personally see nothing wrong with a `require` inside a catch or if/then (switch seems weird, but whatever). It would be nice to have an option that only complained about `require` in new variable scopes.

**Comment by gyandeeps** [comment]
I think the convo above suggests to put something in the docs. is that correct?
CC @nzakas

**Comment by nzakas** [comment]
Yes. We should basically say that this rule mimics the behavior of ES6 modules imports.

**Comment by btmills** [comment]
Working on this.
